src/uring.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2011 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NINJA_URING_H_
+#define NINJA_URING_H_
+
+#include "timestamp.h"
+
+#include <map>
+
+#include <liburing.h>
+#include <sys/stat.h>
+
+/// Helper class for submitting bulk statx() calls to the Linux kernel
+/// via io_uring
+class BulkStat {
+  typedef void (*Callback)(TimeStamp t, const char *error, void *data);
```

Comment:
Please move down after `private:`

Reply:
Can't. It's used by the constructor, and the typedef must be defined before it's used.

Reply:
ah right, my bad.